src/react/ui-elements/Veeam/ VeeamCapacityModal.tsx

```diff
@@ -0,0 +1,143 @@
+import {
+  Icon,
+  Modal,
+  Stack,
+  Text,
+  ToastProvider,
+  Wrap,
+  useToast,
+} from '@scality/core-ui';
+import { Button, Input, Select } from '@scality/core-ui/dist/next';
+import { useState } from 'react';
+import { usePutObjectMutation } from '../../../js/mutations';
+import {
+  GET_CAPACITY_XML_CONTENT,
+  VEEAM_XML_PREFIX,
+  unitChoices,
+} from './VeeamConstants';
+import { useQueryClient } from 'react-query';
+
+type VeeamCapacityModalProps = {
+  isOpen: boolean;
+  close: () => void;
+  bucketName: string;
+  maxCapacity: string;
+};
+
+type Units = 'GiB' | 'TiB' | 'PiB';
+
+export const VeeamCapacityModalInternal = ({
+  isOpen,
+  close,
+  bucketName,
+  maxCapacity,
+}: VeeamCapacityModalProps) => {
+  const [capacity, setCapacity] = useState<string>(maxCapacity.split(' ')[0]);
+  const [capacityUnit, setCapacityUnit] = useState<string>(
+    `${unitChoices[maxCapacity.split(' ')[1] as Units]}`,
+  );
+  const capacityBytes = (
+    parseInt(capacity, 10) * parseInt(capacityUnit, 10)
+  ).toString();
+  const { mutate } = usePutObjectMutation();
+  const { showToast } = useToast();
+  const queryClient = useQueryClient();
+
+  return (
+    <Modal
+      close={close}
+      isOpen={isOpen}
+      footer={
+        <Wrap>
+          <p></p>
+          <Stack>
+            <Button variant="outline" onClick={close} label="Cancel" />
+            <Button
```

Comment:
This button should be disabled if the value didn't change and should be a "submit" button to be semantically correct. To simplify things I would bring use-form)hook and wrap this modal with a form component